src/transformers/models/vit_mae/modeling_tf_vit_mae.py

```diff
@@ -966,11 +966,8 @@ def patchify(self, pixel_values):
         """
         patch_size, num_channels = self.config.patch_size, self.config.num_channels
         # make sure channels are last
-        pixel_values = tf.cond(
-            tf.math.equal(shape_list(pixel_values)[1], num_channels),
-            lambda: tf.transpose(pixel_values, perm=(0, 2, 3, 1)),
-            lambda: pixel_values,
-        )
+        if shape_list(pixel_values)[1] == num_channels:
```

Comment:
I don't think the updated check is compatible with graph mode / compilation of models for XLA

Reply:
`shape_list` is actually quite smart (because I wasn't the one who wrote it) - it returns a list of the static shape for each dimension when this is known at compile time, and the dynamic shape when it isn't. In an XLA compilation pass shapes are fully static, and so the static shape will always be fully known. As a result, all `shape_list` calls will just return the static shape in an XLA context and not introduce data-dependent conditionals.

Reply:
Nice! But also don't put yourself down - I thought you'd written it :) I fear we shall never rid ourselves of `shape_list`